src/main/java/com/indeed/authorization/client/validators/IndeedAccessTokenClaimsVerifier.java

```diff
@@ -0,0 +1,139 @@
+        }
+        this.expectedClientID = clientID;
+
+        if (scopes == null) {
+            throw new IllegalArgumentException("The expected scopes must not be null");
+        }
+        this.expectedScopes = scopes;
+
+        setMaxClockSkew(maxClockSkew);
+    }
+
+    @Override
+    public int getMaxClockSkew() {
+        return this.maxClockSkew;
+    }
+
+    @Override
+    public void setMaxClockSkew(final int maxClockSkewSeconds) {
+        if (maxClockSkewSeconds < 0) {
+            throw new IllegalArgumentException("The max clock skew must be zero or positive");
+        }
+        this.maxClockSkew = maxClockSkewSeconds;
+    }
+
+    @Override
+    public void verify(final JWTClaimsSet claimsSet, final JWKSecurityContext context)
+            throws BadJWTException {
+        try {
+            this.verifyInternal(new IndeedAccessTokenClaimsSet(claimsSet));
+        } catch (final ParseException e) {
+            throw new BadJWTException(e.getMessage(), e);
+        }
+    }
+
+    /**
+     * Verifies selected or all claims from the specified JWT claims set.
+     *
+     * @param claimsSet The JWT claims set. Not null.
+     * @throws BadJWTException If the JWT claims set is rejected.
+     */
+    private void verifyInternal(final IndeedAccessTokenClaimsSet claimsSet) throws BadJWTException {
+        final String iss = claimsSet.getIssuer().getValue();
+        if (!this.isValidIssuer(iss)) {
+            this.throwBadJWTException("Unexpected issuer claim", this.expectedIssuer, iss);
+        }
+
+        final String azp = (String) claimsSet.getClaim(AZP_CLAIM_NAME);
+        if (!isValidAuthorizedParty(azp)) {
+            this.throwBadJWTException(
+                    "Unexpected authorized party claim", this.expectedClientID, azp);
+        }
+
+        final String scope = (String) claimsSet.getClaim(SCOPE_CLAIM_NAME);
+        if (!this.areScopesGranted(scope)) {
+            this.throwBadJWTException(
+                    "Requested scope were not all granted",
+                    Arrays.toString(this.expectedScopes),
+                    scope);
+        }
+
+        final long exp = (long) claimsSet.getClaim(EXP_CLAIM_NAME);
+        if (this.isExpired(exp)) {
+            this.throwBadJWTException("Token is expired", System.currentTimeMillis(), exp);
+        }
+    }
+
+    private boolean isValidIssuer(final String iss) {
+        return this.expectedIssuer.getValue().equals(iss);
+    }
+
+    private boolean isValidAuthorizedParty(final String azp) {
+        return this.expectedClientID.getValue().equals(azp);
+    }
+
+    private boolean areScopesGranted(final String scope) {
+        return new HashSet<>(Arrays.asList(scope.split(SCOPE_CLAIM_DELIMITER)))
+                .containsAll(Arrays.asList(this.expectedScopes));
+    }
+
+    private boolean isExpired(final long exp) {
+        return System.currentTimeMillis() > exp;
+    }
+
+    private <T> void throwBadJWTException(
```

Comment:
nit: It's usually considered as anti-pattern to have a function that always throws an exception. A better way of doing this like this is to write your own exception class and take those parameters in the constructor.

Reply:
This was implemented in [5bc24444dc49c402a3163caff5f4b555fcbcbc36](https://github.com/indeedeng/oauth-client/pull/4/commits/5bc24444dc49c402a3163caff5f4b555fcbcbc36). 

This should have been implemented for what I am trying to acheive.